### PR TITLE
72 configure rtl support, 85 Configuration of Theme Provider for Material-UI

### LINKED ISSUE #72: Configure rtl support
### Main Things : 
- [x] Material-UI Right-to-left Configuration
- [ ] Arabic font from google font

#### Material-UI Right-to-left Configuration
- To change the direction of Material-UI components  the following steps must be followed:
1. HTML
 the dir attribute should be set on the body, otherwise native components will break
```bash
<body dir="rtl">
```
2. jss-rtl
```bash
npm install jss-rtl
```

3.  When  a JSS instance with the plugin is created, it should be available to all the components in the component tree. The StylesProvider component enables this:
```bash
import { create } from 'jss';
import rtl from 'jss-rtl';
import { StylesProvider, jssPreset } from '@material-ui/core/styles';

// Configure JSS
const jss = create({ plugins: [...jssPreset().plugins, rtl()] });

function RTL(props) {
  return (
    <StylesProvider jss={jss}>
      {props.children}
    </StylesProvider>
  );
}
```
#### Arabic font from google font
-  A clear arabic font should be chosen, it should be consistent  with the design and appearance of the website.

[Material-UI Right-to-left Configuration](https://material-ui.com/guides/right-to-left/)

### LINKED ISSUE #85: Configuration of Theme Provider for Material-UI
## Theme configuration variables

-  Wrapping the app around theme provider to give the component that are nested within the application access to any variable within that theme provider

```javascript
import { createMuiTheme ,ThemeProvider}  form '@material-ui/core/styles'

<ThemeProvider theme={theme}>
  <App />
</ThemeProvider>
```
### Palette colors
- Selected theme colors from Figma should be used here as primary, secondary, error, warning, info, and success palette colors
```javascript
import { createMuiTheme ,ThemeProvider}  form '@material-ui/core/styles'

const theme = createMuiTheme({
  palette: {
    primary: {
      main: ''#E24928'
    },
    secondary: {
      main: '#E0E0E0'
    },
  },
}); 

```
- Other mode colors can also be used if needed as main, light, and dark.

#### Font Family

```javascript
const theme = createMuiTheme({
  typography: {
    fontFamily: [
//here use the arabic font from google fonts
      '-apple-system',
      'BlinkMacSystemFont',
      '"Segoe UI"',
      'Roboto',
    ].join(','),
  },
});
```
#### Font Size

```javascript
const theme = createMuiTheme({
  typography: {
    fontSize: 12,
  },
});
```
- There are many other variables, check the link for more details:
[Theming Material-UI](https://material-ui.com/customization/theming/)

### PR DESCRIPTION
**Pull request by Sondos-Alalami**
fixed #72
fixed #85 
####  Configuration of Theme Provider for Material-UI:
- Installation of @material-ui/core.
- Using createMuiTheme to create custom theme variables (colors, breakpoints, typography ).
- Wrapping ThemeProvider around app components.
#### configuration of rtl support:
- modify dir of html to be 'rtl'.
- Adding direction property to theme, and give it a value of 'rtl'.
-  Installation of jss-rtl.
-  Creating StyleProvider to pass jss to the components.

### CODE REVIEW

File: client/public/index.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <!DOCTYPE html>
-<html lang="en">
+<html lang="en" dir="rtl">
```
Comment (AlaaShurrab) [inline client/public/index.html]: is it possible to use lang = ar?
[link](https://www.w3schools.com/tags/ref_language_codes.asp)

**Review by Ahmed-aed — DISMISSED** [review]
god job

File: client/src/App.js [hunk 2]
```diff
@@ -1,8 +1,20 @@
+import { create } from 'jss';
+import rtl from 'jss-rtl';
+import { StylesProvider, jssPreset, ThemeProvider } from '@material-ui/core/styles';
+
+import theme from './component/component1/theme';
```
Comment (mFlaifel) [inline client/src/App.js]: please rename the compont1 folder to something more specific like theme

File: client/src/index.css [hunk 3]
```diff
@@ -1,3 +1,4 @@
+@import url('https://fonts.googleapis.com/css2?family=Harmattan&family=Tajawal&display=swap');
```
Comment (mFlaifel) [inline client/src/index.css:1]: if you wonder like me, what is font-display [this link will help you](https://css-tricks.com/almanac/properties/f/font-display/)

**Review by mFlaifel — APPROVED** [review]

**Review by Ahmed-aed — APPROVED** [review]

**Review by AlaaShurrab — APPROVED** [review]

**Pull request merged**